mne/time_frequency/tfr.py

```diff
@@ -459,111 +680,245 @@ def single_trial_power(data, sfreq, frequencies, use_fft=True, n_cycles=7,
     return power
 
 
-def _induced_power_cwt(data, sfreq, frequencies, use_fft=True, n_cycles=7,
-                       decim=1, n_jobs=1, zero_mean=False):
-    """Compute time induced power and inter-trial phase-locking factor
+# Aux function to reduce redundancy between tfr_morlet and tfr_multitaper
 
-    The time frequency decomposition is done with Morlet wavelets
+def _tfr_aux(method, inst, freqs, decim, return_itc, picks, average,
+             **tfr_params):
+    decim = _check_decim(decim)
+    data = _get_data(inst, return_itc)
+    info = inst.info
+
+    info, data, picks = _prepare_picks(info, data, picks)
+    data = data[:, picks, :]
+
+    if average:
+        if return_itc:
+            output = 'avg_power_itc'
+        else:
+            output = 'avg_power'
+    else:
+        output = 'power'
+        if return_itc:
+            raise ValueError('Inter-trial coherence is not supported'
+                             ' with average=False')
+
+    out = _compute_tfr(data, freqs, info['sfreq'], method=method,
+                       output=output, decim=decim, **tfr_params)
+    times = inst.times[decim].copy()
+
+    if average:
+        if return_itc:
+            power, itc = out.real, out.imag
+        else:
+            power = out
+        nave = len(data)
+        out = AverageTFR(info, power, times, freqs, nave,
+                         method='%s-power' % method)
+        if return_itc:
+            out = (out, AverageTFR(info, itc, times, freqs, nave,
+                                   method='%s-itc' % method))
+    else:
+        power = out
+        out = EpochsTFR(info, power, times, freqs, method='%s-power' % method)
+
+    return out
+
+
+@verbose
+def tfr_morlet(inst, freqs, n_cycles, use_fft=False, return_itc=True, decim=1,
+               n_jobs=1, picks=None, zero_mean=True, average=True,
+               verbose=None):
+    """Compute Time-Frequency Representation (TFR) using Morlet wavelets
```

Comment:
Time Frequency (the oxford rule is no hyphen between two nouns, only between adj & noun, IIRC)
wavelets.